src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -227,23 +241,70 @@ func (h *PromWriteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.metrics.writeSuccess.Inc(1)
 }
 
-func (h *PromWriteHandler) parseRequest(r *http.Request) (*prompb.WriteRequest, *xhttp.ParseError) {
+func (h *PromWriteHandler) parseRequest(
+	r *http.Request,
+) (*prompb.WriteRequest, ingest.WriteOptions, *xhttp.ParseError) {
+	var opts ingest.WriteOptions
+	if v := strings.TrimSpace(r.Header.Get(handler.MetricsTypeHeader)); v != "" {
+		// Allow the metrics type and storage policies to override
+		// the default rules and policies if specified.
+		metricsType, err := storage.ParseMetricsType(v)
+		if err != nil {
+			return nil, ingest.WriteOptions{},
+				xhttp.NewParseError(err, http.StatusBadRequest)
+		}
+
+		// Ensure ingest options specify we are overriding the
+		// downsampling rules with zero rules to be applied (so
+		// only direct writes will be made).
+		opts.DownsampleOverride = true
+		opts.DownsampleMappingRules = nil
+
+		strPolicy := strings.TrimSpace(r.Header.Get(handler.MetricsStoragePolicyHeader))
```

Comment:
will the caller get a reasonable error back if the storage policy they're trying to write to doesn't exist?

Reply:
Yup, they eventually get (have seen this when I've been testing too):
```go
			return result(nil,
				fmt.Errorf("could not find namespace for storage policy: %v",
					restrict.StoragePolicy.String()))
```